client.go

```diff
@@ -268,6 +270,22 @@ func (c *Client) Subscribe(channelName string, opts ...SubscribeOption) (Channel
 	return ch, ch.Subscribe(opts...)
 }
 
+// SubscribePresence creates a subscription to the specified presence channel.
+// If the channel has already been subscribed, this method will return the
+// existing channel instance.
+//
+// An error is returned if channelName is not a presence channel. Use
+// Subscribe() for other channel types.
+func (c *Client) SubscribePresence(channelName string, opts ...SubscribeOption) (PresenceChannel, error) {
+
+	if !strings.HasPrefix(channelName, "presence-") {
+		return nil, fmt.Errorf("invalid presence channel name, must start with 'presence-': %s", channelName)
+	}
+
+	ch, subscribeErr := c.Subscribe(channelName, opts...)
+	return ch.(*presenceChannel), subscribeErr
```

Comment:
I have a suspicion that this may panic when `c.Subscribe` returns `nil, <some error>`. I tried replicating [here](https://go.dev/play/p/fPZ2yf4RWLx), but I'm not sure that's correct example.

Can we pls make it explicit like
```suggestion
	ch, subscribeErr := c.Subscribe(channelName, opts...)
	if err != nil {
		return nil, subscribeErr
	}
	
	// <comment about why this will never panic or another explicit type cast and error handling>
	return ch.(*presenceChannel), nil
```

Reply:
Can we add a test case to confirm this?

Reply:
Hm yeah that playground example indicates this will panic. I'll update it like you mentioned and add test coverage.